components/dada-ir/src/token_tree.rs

```diff
@@ -16,8 +11,11 @@ salsa::entity2! {
 impl<Db: ?Sized + crate::Db> salsa::DebugWithDb<Db> for TokenTree {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>, db: &Db) -> std::fmt::Result {
         let db = db.as_dyn_ir_db();
-        let file_span: FileSpan = self.span(db).in_file(self.filename(db));
-        write!(f, "Tokens({:?})", file_span.into_debug(db))
+        if !f.alternate() {
+            write!(f, "{:?}", self.tokens(db))
+        } else {
+            write!(f, "{:#?}", self.tokens(db))
+        }
```

Comment:
So, I rather intentionally kept token trees from printing all the tokens, because the output is overwhelming. But I do think it's useful to have this output. I'm not sure the best way to balance this, but clearly we need a comment either way.

I *think* I would prefer that you "opt-in" to printing the tokens by e.g. doing `token_tree.details`, but I wonder if what we actually want is some configuration, like rustc's `-Zverbose`?

Reply:
I was wondering if that was the case.